Tests/Unit/ClientEntityTest.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Trikoder\Bundle\OAuth2Bundle\Tests\Unit;
+
+use PHPUnit\Framework\TestCase;
+use Trikoder\Bundle\OAuth2Bundle\Model\Client;
+
+class ClientEntityTest extends TestCase
+{
+    /**
+     * @dataProvider confidentialDataProvider
+     */
+    public function testClientConfidentiality(?string $secret, bool $isConfidential): void
+    {
+        $client = new Client('foo', $secret);
+
+        $this->assertSame($isConfidential, $client->isConfidential());
+    }
+
+    public function confidentialDataProvider(): iterable
+    {
+        return [
+            'Client with null secret is not confidential' => [null, false],
+            'Client with empty secret is not confidential' => ['', false],
```

Comment:
@X-Coder264 Why do we want this case?

Reply:
@HypeMC Passport also behaves in the same way: https://github.com/laravel/passport/blob/v8.4.0/src/Client.php#L103